Fix remaining issues in applications and demos from the updated spaceros base image

eholum opened the pull request:
Resolves #161.

There are still issues building `main` for moveit and nav2 after migration of the base image build to the space-ros repo. This PR fixes them, including

* Updating changed environment variable names
* Fixes references to moved or removed directories in the base image
* Parameterizes the base image name for building locally and in CI

As the base `main` image is now pushed to GHCR, we should use that when building in CI. The change can be tested locally if you have access to the container registry (you _must_ be logged in):

Update:

Depends on https://github.com/space-ros/space-ros/pull/220

Because the GHCR images are not public, we can temporarily point to the main branch in the docker registry. This can be tested with either `moveit2` or `nav2` with:

```
$ docker pull osrf/space-ros:main
$ SPACE_ROS_IMAGE=osrf/space-ros:main ./build.sh
```

asimonov commented:
I am trying to test #152, but that requires moveit2 docker which is failing on `main`.

shall we test/push this one first to make life easier?

asimonov commented:
yes, I am using the locally built images. but it is getting confusing...

basically if our ultimate test is to run moveit and rover demos that requires all the images to 1. be built, 2. be consistent. and they require different PRs to be brought to that state.

if we could save branch tagged images and start merging PRs, that would alleviate confusion to some degree

eholum commented:
> yes, I am using the locally built images. but it is getting confusing...

Ugh, yes sorry things have been in an odd state, but now that the main image is updated and pushed to Dockerhub, too things should be more clear? This PR should resolve the build issues on `main`, and also open things up to using different base images for the moveit and nav2 builds.

asimonov commented:
> > yes, I am using the locally built images. but it is getting confusing...
> 
> Ugh, yes sorry things have been in an odd state, but now that the main image is updated and pushed to Dockerhub, too things should be more clear? This PR should resolve the build issues on `main`, and also open things up to using different base images for the moveit and nav2 builds.

almost... git-lfs change on simulation broke my local build of docker/space_robots... https://github.com/space-ros/simulation/issues/35

i think we need to test every PR on affected repos by running those 2 demos. because they effectively combine everything we have

asimonov commented:
Erik, can you include the lfs setup into this PR as well?

eholum commented:
> Erik, can you include the lfs setup into this PR as well?

Ugh, in the interest of traceability I included that fix in https://github.com/space-ros/docker/pull/194, since that was broken by a separate change... 

> i think we need to test every PR on affected repos by running those 2 demos. because they effectively combine everything we have

This is a big ask until we get some sort of automated infrastructure in place to at least pre-build those images. Or in an ideal worth a pre-build and a basic sanity check. But hopefully we can get that done sooner rather than later?

asimonov reviewed (approved):
looks good. and it builds locally.

but i cannot test demos (mars rover and navigation) because of other issues (e.g. https://github.com/space-ros/docker/pull/194)

i suggest we merge all these PRs releated to fixing the build and then re-test the demos

EzraBrooks reviewed (changes requested):
Can we break out the GitHub Actions version bumps into another issue/PR? Everything else is clearly related

eholum commented:
> Can we break out the GitHub Actions version bumps into another issue/PR? Everything else is clearly related

Done. Squashed those changes out of the param change commits and updated. Everything else remains unchanged. I'll open a separate issue to update CI since there's more that would be helpful now that we have main available...

EzraBrooks reviewed (approved):
These code changes all make obvious sense to me.

eholum commented:
> i suggest we merge all these PRs releated to fixing the build and then re-test the demos

The time is nigh!
